### PR TITLE
Remove final notification and update notification count when Post deleted

### LINKED ISSUE #6651: Deleting a Post should remove any related notifications
### Expected behavior
Deleting/trash a Post in the Posts list should also remove any outstanding Post Upload notifications for such a Post.

### Actual behavior
In particular, if there is an outstanding upload error notification for a Post, and the user decides to delete/trash such Post, the Post is deleted but the upload error notification for such Post remains in their dashbord.

### Steps to reproduce the behavior
1. start a draft
2. place a breakpoint in `uploadPost()` in `UploadService`.
3. insert an image
4. exit the editor
5. when the image is done uploading, the breakpoint will be reached.
6. at this point, turn airplane mode ON
7. let the execution continue
8. a post upload error notification is shown in the user's dashboard.
9. now, trash/delete the Post
10. observe que upload error notification is still there on the users's dashboard.

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #6651 

To test:

**CASE A: remove success Post notification**
1. start a draft
2. include several media items in the Post
3. exit the editor
4. you should see the progress notification read “1 post, xxx media files remaining"
5. send the app to the background so the success notification can be created
6. wait until it all uploads, and see the success notification is shown
7. open the app again, go to the Posts list and tap “Delete” or “Trash” on the post you just uploaded.
8. once the UNDO snackbar appears, let it disappear
9. observe the final success notification for such a Post is gone.

**CASE B: remove error Post notification**
1. start a draft
2. include several media items in the Post
3. exit the editor
4. you should see the progress notification read “1 post, xxx media files remaining"
5. turn airplane mode ON
6. wait a bit and the progress notification disappears, and an error notification appears.
7. open the app again, go to the Posts list and tap “Delete” or “Trash” on the post you just uploaded.
8. once the UNDO snackbar appears, let it disappear
9. observe the final error notification for such a Post is gone.

**CASE C: decrement count from progress notification on the go**
1. start a draft
2. include several media items in the Post (the more the better so we can have 2 Posts simultaneously uploaded)
3. exit the editor
4. you should see the progress notification read “1 post, xxx media files remaining"
5. repeat steps 1 through 3, you should see the progress notification read “2 posts, xxx media files remaining”
6. before the first Post finishes uploading, go to the Posts list and tap “Delete” or “Trash” on the post.
8. once the UNDO snackbar appears, let it disappear
9. observe the progress notification shows now “1 post, xxx media files remaining”
10. let it finish and observe the success notification for the remaining Post (if you send the app to the background before it finishes)

cc @nbradbury

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java [hunk 1]
```diff
@@ -146,6 +146,17 @@ void addPostInfoToForegroundNotification(@NonNull PostModel post, @Nullable List
         startOrUpdateForegroundNotification(post);
     }
 
+    void removePostInfoFromForegroundNotification(@NonNull PostModel post, @Nullable List<MediaModel> media) {
+        sNotificationData.totalPostItems--;
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java]: Can we call `removePostInfoFromForegroundNotificationData` here?
It seems almost the the same code...


Also, not related to this PR, what do you think on remove the method
```
    void prepareForegroundNotificationForRetry(@NonNull PostModel post, @Nullable List<MediaModel> media) {
        removePostInfoFromForegroundNotificationData(post, media);
    }
```

and if you think is necessary rename `removePostInfoFromForegroundNotificationData`  ?

Not sure why we have a method that does nothing, except to call another method with the same signature.

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java]: thank you for noticing 👀 
addressed in d554953

**Comment by daniloercoli** [comment]
**CASE B: remove error Post notification**

Media that were attached to the post, ant not yet uploaded are in the media library with the failed status. Don't think it's related to this PR, but I think we should remove it from the media library.

**Comment by daniloercoli** [comment]
I was testing this PR and found the following case. Call it **Case D**

- start a draft
- include several media items in the Post
- exit the editor
- you should see the progress notification read “1 post, xxx media files remaining"
- Close the app from the system manager
- You will see double notification. One with the error and one with the ongoing uploading

screenshot_20171027-102749

- This is what you see in post lists
screenshot_20171027-102820


- And the uploading is actually in progress
screenshot_20171027-102808


I'm also not sure why there is the following label in it: "6 posts".

**Comment by mzorz** [comment]
Nice finds @daniloercoli ! Going to move these reports to the master PR for foreground notifications here https://github.com/wordpress-mobile/WordPress-Android/pull/6713

**Comment by mzorz** [comment]
Comments ready https://github.com/wordpress-mobile/WordPress-Android/pull/6713#issuecomment-339978185  and https://github.com/wordpress-mobile/WordPress-Android/pull/6713#issuecomment-339979558

Please continue to review this PR with a focus on the 3 test cases described above as this PR only covers updating the notifications when you delete a Post 👍 @daniloercoli

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**